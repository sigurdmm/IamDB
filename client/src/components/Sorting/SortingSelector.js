import PropTypes from 'prop-types';
import React from 'react';

const SortingSelector = ({
  sortingMethods,
  onSort,
  fieldValue,
}) => <>
    <select
      name="select"
      className='sortingbar__type'
      onChange={event => onSort(event.target.value)}
      value = {fieldValue}
    >
    {sortingMethods.map((method, i) => (
      <option
        key={i}
        value={method.value}
      >
        {method.label}
      </option>))}
    </select>
  </>;

SortingSelector.propTypes = {
  onSort: PropTypes.func.isRequired,
  fieldValue: PropTypes.string.isRequired,
  sortingMethods: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  })).isRequired,
};

export default SortingSelector;
