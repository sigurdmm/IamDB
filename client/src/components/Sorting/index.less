@import '../../constants';

.sortingbar {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  margin-bottom: 10px;

  &__hidden {
    display: none;
  }

  &__direction,
  &__type {
    height: 25px;
    border-color: @borderColor;
    font-family: inherit;
    -webkit-border-radius: 2px;
    -moz-border-radius: 2px;
    border-radius: 2px;
    background-color: @white;
  }

  &__direction {
    width: 25px;
    padding: 0;

  }

  &__type {
    width: 200px;
    margin-right: 5px;
    font-size: inherit;
  }

}
