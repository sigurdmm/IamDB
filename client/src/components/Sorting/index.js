import React from 'react';
import PropTypes from 'prop-types';
import SortingSelector from './SortingSelector';
import './index.less';

const Sorting = ({
  directionValue,
  onDirectionClick,
  sortingMethods,
  onSort,
  fieldValue,
}) => <div className='sortingbar'>
    <SortingSelector
      sortingMethods={sortingMethods}
      onSort={onSort}
      fieldValue={fieldValue}
    />
    <button
      className='sortingbar__direction'
      value={directionValue}
      onClick={() => {
        onDirectionClick(directionValue === -1 ? 1 : -1);
      }}
    >
      {directionValue === -1 ? '▼' : '▲'}
    </button>
  </div>;

Sorting.propTypes = {
  directionValue: PropTypes.number.isRequired,
  fieldValue: PropTypes.string.isRequired,
  onDirectionClick: PropTypes.func.isRequired,
  onSort: PropTypes.func.isRequired,
  sortingMethods: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  })).isRequired,
};

export default Sorting;
