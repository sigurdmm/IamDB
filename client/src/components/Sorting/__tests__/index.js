import { shallow } from 'enzyme';

import Sorting from '../index';


describe('<Sorting/>', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Sorting
      directionValue={1}
      onSort={jest.fn()}
      onDirectionClick={jest.fn()}
      fieldValue={''}
      sortingMethods={ [{ label: 'label', value: 'value' }, { label: 'label2', value: 'value2' }] }
      visible={true}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call onDirectionClick on click', () => {
    const spy = jest.fn();

    const wrapper = shallow(<Sorting
        directionValue={1}
        onSort={jest.fn()}
        onDirectionClick={spy}
        fieldValue={''}
        sortingMethods={ [{ label: 'label', value: 'value' }] }
        visible={true}
      />);

    wrapper.find('button').simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
