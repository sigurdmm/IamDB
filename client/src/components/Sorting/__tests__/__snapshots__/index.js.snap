// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sorting/> should render properly 1`] = `
<div
  className="sortingbar"
>
  <SortingSelector
    fieldValue=""
    onSort={[MockFunction]}
    sortingMethods={
      Array [
        Object {
          "label": "label",
          "value": "value",
        },
        Object {
          "label": "label2",
          "value": "value2",
        },
      ]
    }
  />
  <button
    className="sortingbar__direction"
    onClick={[Function]}
    value={1}
  >
    â–²
  </button>
</div>
`;
